dev/io.openliberty.security.jakartasec.3.0.internal/src/io/openliberty/security/jakartasec/tokens/IdentityTokenImpl.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.security.jakartasec.tokens;
+
+import java.io.Serializable;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.BiConsumer;
+
+import jakarta.security.enterprise.identitystore.openid.IdentityToken;
+import jakarta.security.enterprise.identitystore.openid.JwtClaims;
+
+public class IdentityTokenImpl implements IdentityToken, Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private final String tokenString;
+    private final Map<String, Object> idTokenClaimsMap;
+    private final JwtClaims jwtClaims;
+    private final Long tokenMinValidityInMillis;
+
+    public IdentityTokenImpl(String tokenString, Map<String, Object> idTokenClaimsMap, Long tokenMinValidityInMillis) {
+        this.tokenString = tokenString;
+        this.idTokenClaimsMap = idTokenClaimsMap;
```

Comment:
I can see it being an issue if working with JWE tokens, but is the `idTokenClaimsMap` arg necessary if we could just do essentially what you have in the unit test?
```suggestion
        String[] parts = tokenString.split(("\\."));
        String tokenClaimsJsonString = new String(Base64.getDecoder().decode(parts[1]), "UTF-8");
        this.idTokenClaimsMap = org.jose4j.jwt.JwtClaims.parse(tokenClaimsJsonString).getClaimsMap();
```
I'm wondering if there'd be any room for discrepancy between the `tokenString` and the `idTokenClaimsMap`.

Reply:
The TokenResponseValidator already parses the id token and get jose4j's JwtClaims object, 

```
JwtClaims jwtClaims = null;
        if (jwtcontext != null && jwtcontext.getJwtClaims() != null) {
            jwtClaims = jwtcontext.getJwtClaims();
```

so instead of parsing the token again here, the validator can create this IdentityTokenImpl object with the map from the JwtClaims object after successful token verification. That would improve performance.

I agree it might be an issue for JWE is the validator is not able to handle JWE.